utils/dom.js

```diff
@@ -10,16 +10,30 @@ import tinymce from 'tinymce';
 const { getComputedStyle, DOMRect } = window;
 const { TEXT_NODE, ELEMENT_NODE } = window.Node;
 
+/**
+ * Returns true if there is a selection and if the selection is collapsed, or
+ * false otherwise.
+ *
+ * @return {boolean} Whether there is a collapsed selection.
+ */
+export function hasCollapsedSelection() {
```

Comment:
Will this work on `textarea` and `input`? Should a `container` be provided just like `isHorizontalEdge` etc.?

Reply:
That's a good point, and we never actually seem to check if the selection is within the `container` in case of an `contenteditable` field. I do think we should provide the container (needed at least of `textarea` and `input`) and ensure _that_ contains a collapsed selection, not just the document in general? Or do you view this differently?

Reply:
E.g. you might argue that since this is triggered by a keydown event, we can assume the selection is in the `container`.

Reply:
>not just the document in general

My thinking was toward the document selection in general. Like `container = document.activeElement`.